src/participant/prompts/namespace.ts

```diff
@@ -39,44 +47,33 @@ User: Where is the best hummus in Berlin?
 Response:
 No names found.
 `;
-
-    // eslint-disable-next-line new-cap
-    return vscode.LanguageModelChatMessage.Assistant(prompt);
   }
 
-  static getUserPrompt(prompt: string): vscode.LanguageModelChatMessage {
-    // eslint-disable-next-line new-cap
-    return vscode.LanguageModelChatMessage.User(prompt);
+  protected getUserPrompt({ request }: NamespacePromptArgs): Promise<string> {
+    return Promise.resolve(request.prompt);
   }
 
-  static buildMessages({
+  async buildMessages({
```

Comment:
This one is the only one I couldn't map super cleanly and had to override the base class implementation. I have some ideas for how to restructure the abstraction to support these more complex history/prompt rewrites, but those would be more complicated than the current implementation and didn't feel it's justified to pursue until we have more than one use case.

Reply:
We'll revisit this function and also how the other prompt message building functions handle namespace and connect messages/prompts in:
https://jira.mongodb.org/browse/VSCODE-611 
I think the other prompts would benefit from a similar prompt change when the last message was selecting a namespace.

Reply:
Agreed - I was following up on this PR with telemetry stuff and noticed history scraping could be applied to all messages, not just namespace.